geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/RegisterInterestOnServerWithoutRegionRegressionTest.java

```diff
@@ -0,0 +1,158 @@
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertNotNull;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.RegionDestroyedException;
+import org.apache.geode.cache.client.ClientRegionFactory;
+import org.apache.geode.cache.client.ClientRegionShortcut;
+import org.apache.geode.cache.client.PoolManager;
+import org.apache.geode.cache.client.ServerOperationException;
+import org.apache.geode.cache.client.internal.PoolImpl;
+import org.apache.geode.cache.server.CacheServer;
+import org.apache.geode.test.dunit.IgnoredException;
+import org.apache.geode.test.dunit.VM;
+import org.apache.geode.test.dunit.rules.CacheRule;
+import org.apache.geode.test.dunit.rules.ClientCacheRule;
+import org.apache.geode.test.dunit.rules.DistributedTestRule;
+import org.apache.geode.test.junit.categories.ClientServerTest;
+import org.apache.geode.test.junit.categories.DistributedTest;
+import org.apache.geode.test.junit.rules.serializable.SerializableTestName;
+
+/**
+ * When server is running but region is not created on server. Client sends register interest
+ * request, server checks for region, and if region is not exist on server, it throws an exception
+ * to the client. Hence, client marks server as dead.
+ *
+ * <p>
+ * To avoid this, there should not be any check of region before registration. And region
+ * registration should not fail due to non existent region.
+ *
+ * <p>
+ * TRAC #36805: Interest registration on server should not check for region in server's cache.
+ */
+@Category({DistributedTest.class, ClientServerTest.class})
+public class RegisterInterestOnServerWithoutRegionRegressionTest implements Serializable {
+
+  private String uniqueName;
+  private String regionName;
+  private String hostName;
+
+  private VM server;
+  private VM client;
+
+  @Rule
+  public DistributedTestRule distributedTestRule = new DistributedTestRule();
+
+  @Rule
+  public CacheRule cacheRule = new CacheRule();
+
+  @Rule
+  public ClientCacheRule clientCacheRule = new ClientCacheRule();
+
+  @Rule
+  public SerializableTestName testName = new SerializableTestName();
+
+  @Before
+  public void setUp() throws Exception {
+    server = getVM(0);
+    client = getVM(1);
+
+    uniqueName = getClass().getSimpleName() + "_" + testName.getMethodName();
+    regionName = uniqueName + "_region";
+    hostName = getHostName();
+
+    int port = server.invoke(() -> createServerCache());
+
+    client.invoke(() -> createClientCache(port));
+
+    client.invoke(() -> awaitConnectedServerCount(1));
+  }
+
+  @Test
+  public void registerInterestDoesNotRequireRegionOnServer() {
```

Comment:
I like `Does` (as in this name) way better than `Should` (as in some of the other names).

Reply:
I updated some class test names and test method names.